metadata-ingestion/tests/integration/powerbi/test_stateful_ingestion.py

```diff
@@ -0,0 +1,224 @@
+        },
+        "https://api.powerbi.com/v1.0/myorg/groups/64ED5CAD-7C10-4684-8180-826122881108/dashboards/e41cbfe7-9f54-40ad-8d6a-043ab97cf303/tiles": {
+            "method": "GET",
+            "status_code": 200,
+            "json": {"value": []},
+        },
+    }
+
+    for url in api_vs_response.keys():
+        request_mock.register_uri(
+            api_vs_response[url]["method"],
+            url,
+            json=api_vs_response[url]["json"],
+            status_code=api_vs_response[url]["status_code"],
+        )
+
+
+def register_mock_api_state2(request_mock):
+    api_vs_response = {
+        "https://api.powerbi.com/v1.0/myorg/groups": {
+            "method": "GET",
+            "status_code": 200,
+            "json": {
+                "value": [
+                    {
+                        "id": "64ED5CAD-7C10-4684-8180-826122881108",
+                        "isReadOnly": True,
+                        "name": "Workspace 1",
+                        "type": "Workspace",
+                    },
+                ]
+            },
+        },
+        "https://api.powerbi.com/v1.0/myorg/groups/64ED5CAD-7C10-4684-8180-826122881108/dashboards": {
+            "method": "GET",
+            "status_code": 200,
+            "json": {
+                "value": [
+                    {
+                        "id": "7D668CAD-7FFC-4505-9215-655BCA5BEBAE",
+                        "isReadOnly": True,
+                        "displayName": "marketing",
+                        "embedUrl": "https://localhost/dashboards/embed/1",
+                        "webUrl": "https://localhost/dashboards/web/1",
+                    }
+                ]
+            },
+        },
+        "https://api.powerbi.com/v1.0/myorg/groups/64ED5CAD-7C10-4684-8180-826122881108/dashboards/7D668CAD-7FFC-4505-9215-655BCA5BEBAE/tiles": {
+            "method": "GET",
+            "status_code": 200,
+            "json": {"value": []},
+        },
+    }
+
+    for url in api_vs_response.keys():
+        request_mock.register_uri(
+            api_vs_response[url]["method"],
+            url,
+            json=api_vs_response[url]["json"],
+            status_code=api_vs_response[url]["status_code"],
+        )
+
+
+def default_source_config():
+    return {
+        "client_id": "foo",
+        "client_secret": "bar",
+        "tenant_id": "0B0C960B-FCDF-4D0F-8C45-2E03BB59DDEB",
+        "workspace_id": "64ED5CAD-7C10-4684-8180-826122881108",
+        "extract_lineage": False,
+        "extract_reports": False,
+        "enable_admin_api": False,
+        "extract_ownership": False,
+        "stateful_ingestion": {
+            "enabled": True,
+        },
+        "convert_lineage_urns_to_lowercase": False,
+        "workspace_id_pattern": {"allow": ["64ED5CAD-7C10-4684-8180-826122881108"]},
+        "dataset_type_mapping": {
+            "PostgreSql": "postgres",
+            "Oracle": "oracle",
+        },
+        "env": "DEV",
```

Comment:
this config seems like it needs something similar to this to make the tests work https://github.com/datahub-project/datahub/blob/667ca8632d7e86fd2e3d1c4d555ce0e9544dd971/metadata-ingestion/tests/integration/lookml/test_lookml.py#L533-L535